tests/aws/services/lambda_/test_lambda.py

```diff
@@ -1146,6 +1146,75 @@ def test_lambda_url_invocation_exception(self, create_lambda_function, snapshot,
         assert to_str(result.content) == "Internal Server Error"
         assert result.status_code == 502
 
+    @markers.aws.validated
+    def test_lambda_url_persists_after_alias_delete(
+        self, create_lambda_function, snapshot, aws_client
+    ):
+        snapshot.add_transformer(
+            snapshot.transform.key_value(
+                "FunctionUrl", "<function-url>", reference_replacement=False
+            )
+        )
+        function_name = f"test-fnurl-echo-{short_uid()}"
+        alias_name = f"alias-{short_uid()}"
+
+        create_lambda_function(
+            func_name=function_name,
+            handler_file=TEST_LAMBDA_URL,
+            runtime=Runtime.nodejs20_x,
+        )
+
+        fn_version_v1 = aws_client.lambda_.publish_version(FunctionName=function_name).get(
+            "Version"
+        )
+
+        aws_client.lambda_.create_alias(
+            FunctionName=function_name, FunctionVersion=fn_version_v1, Name=alias_name
+        )
+
+        aws_client.lambda_.get_waiter(waiter_name="function_updated_v2").wait(
+            FunctionName=function_name, Qualifier=alias_name
+        )
+
+        # Create an aliased URL
+        url_config = aws_client.lambda_.create_function_url_config(
+            FunctionName=function_name,
+            AuthType="NONE",
+            InvokeMode=InvokeMode.BUFFERED,
+            Qualifier=alias_name,
+        )
+        snapshot.match("create_lambda_url_config", url_config)
+
+        aws_client.lambda_.get_waiter(waiter_name="function_active_v2").wait(
+            FunctionName=function_name, Qualifier=alias_name
+        )
+
+        delete_alias_resp = aws_client.lambda_.delete_alias(
+            FunctionName=function_name, Name=alias_name
+        )
+        snapshot.match("delete_alias_resp", delete_alias_resp)
+
+        # Confirm the alias has deleted successfully
+        def _assert_alias_deleted():
+            assert (
+                len(
+                    aws_client.lambda_.list_aliases(
+                        FunctionName=function_name, FunctionVersion=fn_version_v1
+                    ).get("Aliases")
+                )
+                == 0
+            )
+
+        retry(_assert_alias_deleted, retries=15)
```

Comment:
Interesting, so `delete_alias` happens asynchronously, but there is no waiter (unlike for `create_alias` above)?

Reply:
We get a `204` if successful -- the docs don't mention whether this is eventual or not. In truth, I wasn't sure whether this was async or not. Just wanted to be sure things actually did clear up to try minimise flakiness. Maybe I'm being pedantic here with this check.

Reply:
I would try matching the AWS behavior in the test because async behavior is often harder to test and maintain. If a test uses retries, I assume the behavior is async (unless there's an appropriate note).